tabs/tab_corr.py

```
@@ -460,19 +488,99 @@ def out_corr_table():
 
-        fig = result['figure']
-        html_str = fig.to_html(full_html=False, include_plotlyjs='cdn')
-        return ui.HTML(html_str)
+        return result['figure']  # ✅ Return figure directly
+    
+    # ✅ CHANGED: Logic for downloading file
+    @render.download(
+        filename=lambda: f"correlation_{corr_result.get()['var1']}_{corr_result.get()['var2']}.html" 
+        if corr_result.get() else "correlation_report.html"
+    )
+    def btn_dl_corr():
```

Ruff (0.14.10)

Missing return type annotation for private function `btn_dl_corr`

(ANN202)